### ISSUE TITLE
Infinite resource saving

### ISSUE THREAD

**Issue by rutkas** [modxcms/revolution#15004]
When saving resource it is infinite saving progress bar and error in browser console:

```
Uncaught SyntaxError: Unexpected identifier
    at doDecode (ext-all.js:21)
    at Object.decode (ext-all.js:21)
    at Ext.form.Action.Submit.handleResponse (modx.jsgrps-min.js:1)
    at Ext.form.Action.Submit.processResponse (ext-all.js:21)
    at Ext.form.Action.Submit.success (ext-all.js:21)
    at o (ext-all.js:21)
    at Ext.data.Connection.s (ext-all.js:21)
    at HTMLIFrameElement.I (ext-all.js:21)
```
After investigation of the problem, I found that `&quot;` in title field of the PARENT resource caused such a problem with saving any of its child.

### Step to reproduce
Create resource with ` & q u o t ;` in title field.
Create child resource in it and save it.

### Environment
Checked on Latest Chrome, Firefox.
Server version: Apache/2.4.41 (Unix)
PHP Version 7.3.14
MODX Revolution 2.7.2-pl

**Comment by JoshuaLuckers** [comment]
I'm unable to reproduce this using Safari. I tested it in the latest development versions of MODX 2.7.x and 3.x.

**Comment by rutkas** [comment]
I can not say about dev version, but I can reproduce the bug on the latest release (2.7.2), tried also edge and safari on ios - they have this bug too. so I think it is not browser related.

**Comment by ghost** [comment]
No problem on the MODX Cloud using Firefox.
<img width="363" alt="Screen Shot 2020-02-16 at 22 07 11" src="https://user-images.githubusercontent.com/1073780/74611881-ba92ea00-5108-11ea-9ecf-8ec2b3d734cd.png">

**Comment by Jako** [comment]
I could reproduce it here, when you save the resource for the 2nd time (the initial save works fine). The `&quot;` value occurs in 'parent-cmb' and Chrome can't decode that json response.

**Comment by rutkas** [comment]
> No problem on the MODX Cloud using Firefox.
> <img alt="Screen Shot 2020-02-16 at 22 07 11" width="363" src="https://user-images.githubusercontent.com/1073780/74611881-ba92ea00-5108-11ea-9ecf-8ec2b3d734cd.png">

So if you Open Testing Two (20) document and Save it... do you no error in browser console ?

**Comment by rutkas** [comment]
> I could reproduce it here, when you save the resource for the 2nd time (the initial save works fine). The `&quot;` value occurs in 'parent-cmb' and Chrome can't decode that json response.

:) yes, and not only in Chrome, i checked it also in Firefox, Edge, Safai ios - same behavior

**Comment by Ruslan-Aleev** [comment]
Perhaps it depends on the user's operating system, I use Windows and the bug repeats.
How it looks:

![quote](https://user-images.githubusercontent.com/12523676/74615374-9e5b7100-5139-11ea-87d3-adb0b3eb9d8f.gif)

Although my console has slightly different messages:

```
Uncaught SyntaxError: Unexpected number
    at doDecode (ext-all.js:21)
    at Object.decode (ext-all.js:21)
    at Ext.form.Action.Submit.handleResponse (utilities.js:340)
    at Ext.form.Action.Submit.processResponse (ext-all.js:21)
    at Ext.form.Action.Submit.success (ext-all.js:21)
    at o (ext-all.js:21)
    at Ext.data.Connection.s (ext-all.js:21)
    at HTMLIFrameElement.I (ext-all.js:21)
```

**Comment by Ruslan-Aleev** [comment]
There is also a bug in MODX3.
And in breadcrumbs quote are displayed.

![m3_quote](https://user-images.githubusercontent.com/12523676/74615467-8d5f2f80-513a-11ea-8d5e-f3f70720f9e8.png)

**Comment by ghost** [comment]
You are correct. First save on creation is fine, saving the child resource again fails.

**Comment by Jako** [comment]
If i am right, the parent-cmb value is only used for filling the xtype modx-field-parent-change with the initial value. That should be done by the combo box itself.

**Comment by Ibochkarev** [comment]
Still happens on 3.x. The second save is the one that fails, as Jako described.

The resource form posts with `fileUpload: true`, so Ext reads the connector body from an iframe with `doc.body.innerHTML` (`doFormUpload` in `ext-all-debug.js`). That parse turns the characters `&quot;` inside a JSON string into a raw quote, and `Ext.decode` throws. The progress bar stays up.

`modx-field-parent-change` is named `parent-cmb`. After the update page loads, `MODx.panel.Resource.setup()` sets its value to `parent_pagetitle (id)`. `parent_pagetitle` is `stripTags()` of the parent pagetitle in `manager/controllers/default/resource/update.class.php`, so a title that contains `&quot;` is copied into the field. xPDO runs with `OPT_HYDRATE_ADHOC_FIELDS`, so `Resource/Update` does `fromArray($this->getProperties())` and stores that posted string on the object. `cleanup()` returns it, and the iframe breaks on that value.

The hidden `parent` input already sends the numeric id, and `handleParent()` reads that. Set `submitValue: false` on the parent field so the title stays on screen and is left out of the POST. Unset `parent-cmb` before `fromArray` in `Resource/Update` so a display string cannot come back in the JSON.

When the connector response is not an XHR, wrap the JSON in a textarea. Ext already reads `textarea.value` in `doFormUpload`. `htmlspecialchars` the JSON inside that textarea.

Breadcrumbs in `getPageHeader()` (`modx.panel.resource.js` and `modx.panel.resource.data.js`) print `parents[i].pagetitle` as HTML. Run those labels through `Ext.util.Format.htmlEncode`.
